pkg/authorization/workspace_content_authorizer.go

```diff
@@ -75,10 +87,20 @@ type workspaceContentAuthorizer struct {
 func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
 	cluster, err := genericapirequest.ValidClusterFrom(ctx)
 	if err != nil {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			WorkspaceContentAuditDecision, DecisionNoOpinion,
+			WorkspaceContentAuditReason, fmt.Sprintf("error getting cluster from request: %v", err),
+		)
 		return authorizer.DecisionNoOpinion, WorkspaceAcccessNotPermittedReason, err
 	}
 	// empty or non-root based workspaces have no meaning in the context of authorizing workspace content.
 	if cluster == nil || cluster.Name.Empty() || !cluster.Name.HasPrefix(tenancyv1alpha1.RootCluster) {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			WorkspaceContentAuditDecision, DecisionNoOpinion,
+			WorkspaceContentAuditReason, "empty or non root workspace",
```

Comment:
let's print out the cluster name using %q

Reply:
tenancy.kcp.dev/workspace annotation has it?

Reply:
ah, perfect then 👍 just making sure we're having it in the audit log.